docs/sql/json.md

```diff
@@ -100,20 +80,89 @@ SELECT CAST(
       ] AS VARIANT)
 ```
 
+### `TO_JSON`
 
-### `UNPARSE_JSON`
-
-`PARSE_JSON` converts a `VARIANT` value to a `VARCHAR`.  The
-assumption is that the `VARIANT` was has the structure as produced by
-the `PARSE_JSON` function:
+`TO_JSON` converts a `VARIANT` value to a `VARCHAR`:
 
 - the `VARIANT` `null` value is converted to the string `null`
 - a `VARIANT` wrapping a Boolean value is converted to the respective Boolean string `true` or `false`
 - a `VARIANT` wrapping any numeric value (`DECIMAL`, `TINYINT`, `SMALLINT`, `INTEGER`, `BIGINT`, `REAL`, `DOUBLE`, `DECIMAL`) is converted to the string representation of the value as produced using a `CAST(value AS VARCHAR)`
 - a `VARIANT` wrapping a `VARCHAR` value is converted to a string with double quotes, and with escape sequences for special characters (e.g., quotes), as mandated by the JSON grammar
 - a `VARIANT` wrapping an `ARRAY` with elements of any type is converted to a JSON array, and the elements are recursively converted
 - a `VARIANT` wrapping a `MAP` whose keys have any SQL `CHAR` type, or `VARIANT` values wrapping `CHAR` values will generate a JSON object, by recursively converting each key-value pair.
-- any other data value is a conversion error, and causes the `UNPARSE_JSON` function to produce a `NULL` result
+- a `VARIANT` wrapping a `DATE`, `TIME`, or `DATETIME` value will be serialized as a JSON string
+
+## Processing JSON data using `VARIANT`
+
+The `VARIANT` type enables efficient JSON processing in SQL.  In this sense it is similar to
+the `JSONB` type in Postgres and other databases.  There are two ways to convert
+JSON data to and from `VARIANT`:
+
+1. Use `PARSE_JSON` and `TO_JSON` functions to convert strings to `VARIANT` and back.
+2. Automatically, when ingesting data to or outputting data from columns of type `VARIANT`.
+
+The following example demonstrates the first approach. Here, input events
+contain a field called `json` of type string, which carries JSON-encoded data.
+We ingest this field as a string and use `PARSE_JSON` to convert it to a
+`VARIANT` and store the result in an intermediate view.
+
+```sql
+CREATE TABLE json (id INT, json VARCHAR);
+CREATE VIEW parsed_json AS SELECT id, PARSE_JSON(json) AS json FROM json;
+```
+
+Input events can use any [supported data format](/docs/formats/).  For instance, when
+ingesting a [JSON stream](/docs/formats/json), a valid input record could look like this
+(note the use of escaping in the `json` field):
+
+```json
+{"id": 123, "json": "{\"foo\": \"bar\"}"}
+```
+
+The second approach eliminates the need for the intermediate view by
```

Comment:
it doesn't have to be a view, in general it's just a select that calls parse_json.